### PR TITLE
USB:  cdc_acm: Fixes sample

### PR DESCRIPTION
**Pull request by finikorg**
Write all data in a loop fixing remaining data missing for the sample.

### CODE REVIEW

**Comment by carlescufi** [comment]
CC @IvanSanchez

**Comment by carlescufi** [comment]
CC @sundar-subramaniyan

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Just comment to the commit message, it is confusing, that with 4 bytes only affects QUARK_SE, https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/class/cdc_acm.c#L402

**Comment by finikorg** [comment]
Updated commit message

File: samples/subsys/usb/cdc_acm/src/main.c [hunk 1]
```diff
@@ -43,10 +43,17 @@ static void write_data(struct device *dev, const char *buf, int len)
...
-	uart_fifo_fill(dev, (const u8_t *)buf, len);
-	while (data_transmitted == false)
-		;
+	while (len) {
+		int written;
+
+		data_transmitted = false;
+		written = uart_fifo_fill(dev, (const u8_t *)buf, len);
+		while (data_transmitted == false)
+			;
```
Comment (sundar-subramaniyan) [inline samples/subsys/usb/cdc_acm/src/main.c]: Should a k_yield() be called here? It might give chance to other threads to run.

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**